### PR TITLE
MOEN-25555: Android Unity Wrapper Update And add Geofence Unity Wrapper module

### PR DESCRIPTION
**Pull request by rishabh-engg-sdk**
## Jira Ticket

https://moengagetrial.atlassian.net/browse/MOEN-25555

## Description
Android Unity Wrapper Update And add Geofence Unity Wrapper module

### CODE REVIEW

File: Android/android-unity-wrapper/scripts/release.gradle [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (umangmoe) [inline Android/android-unity-wrapper/scripts/release.gradle]: use `publishing-plugin` built internally instead

File: Android/android-unity-wrapper/unity-wrapper/src/main/java/com/moengage/unity/wrapper/MoEInitializer.kt [hunk 2]
```diff
@@ -37,30 +37,44 @@ import com.moengage.plugin.base.internal.PluginInitializer
...
+     *
+     * @param context Context
+     * @param builder Instance of [MoEngage.Builder]
+     * @since TODO
+     */
+    fun initialiseDefaultInstance(context: Context, builder: MoEngage.Builder) {
         try {
-            Logger.print { "$tag initialize() : Initialising MoEngage SDK." }
-            initialize(context, builder, SdkState.ENABLED)
+            initialiseDefaultInstance(context, builder, SdkState.ENABLED)
```
Comment (umangmoe) [inline Android/android-unity-wrapper/unity-wrapper/src/main/java/com/moengage/unity/wrapper/MoEInitializer.kt]: directly call the plugin base method

File: Android/android-unity-wrapper/unity-library/build.gradle [hunk 3]
```diff
@@ -1,13 +1,12 @@
 plugins {
-  id("com.android.library")
+  alias(moengageInternal.plugins.plugin.android.lib)
 }
 
 android {
-  compileSdkVersion(SdkBuildConfig.compileSdkVersion)
-
+  compileSdk 33
```
Comment (umangmoe) [inline Android/android-unity-wrapper/unity-library/build.gradle]: use common config

**Review by umangmoe — CHANGES_REQUESTED** [review]

File: Android/android-unity-wrapper/scripts/release.gradle (hunk 1 above)
Comment (rishabh-engg-sdk) [reply Android/android-unity-wrapper/scripts/release.gradle]: Done

File: Android/android-unity-wrapper/unity-wrapper/src/main/java/com/moengage/unity/wrapper/MoEInitializer.kt (hunk 2 above)
Comment (rishabh-engg-sdk) [reply Android/android-unity-wrapper/unity-wrapper/src/main/java/com/moengage/unity/wrapper/MoEInitializer.kt]: Done

File: Android/android-unity-wrapper/unity-library/build.gradle (hunk 3 above)
Comment (rishabh-engg-sdk) [reply Android/android-unity-wrapper/unity-library/build.gradle]: Done

**Pull request merged**